contractcourt/htlc_incoming_contest_resolver.go

```diff
@@ -50,6 +51,20 @@ func (h *htlcIncomingContestResolver) Resolve() (ContractResolver, error) {
 		return nil, nil
 	}
 
+	// First try to parse the payload. If that fails, we can stop resolution
+	// now.
+	payload, err := h.decodePayload(h.htlc)
+	if err != nil {
+		log.Debugf("ChannelArbitrator(%v): cannot decode payload of "+
+			"htlc %v", h.ChanPoint, h.HtlcPoint())
+
+		// If we've locked in an htlc with an invalid payload on our
```

Comment:
how could we lock in such an HTLC?

Reply:
crash before we reached `processRemoteAdds` in the link

Reply:
So we will lock in the HTLC, crash, restart and forget about it? Shouldn't we either forward or cancel it in that case?

Just seems a bit dangerous to ignore a locked in HTLC without returning any error here...

Reply:
We can forget about it, because the onion payload is invalid. In the off-chain flow nothing would have happened either.

Reply:
if the payload is invalid we would have canceled the htlc back offchain. now that we are on chain we can't cancel (or decode the payload and figure out what needs to be done) so we just let the incoming party timeout the htlc